### ISSUE TITLE
self contain app

### ISSUE THREAD

**Issue by ypriverol** [compomics/ThermoRawFileParser#82]
@caetera @nielshulstaert can we explore how to release self contain an application for the converter

**Comment by caetera** [comment]
.NET Core has option to publish self-contained executable - https://docs.microsoft.com/en-us/dotnet/core/deploying/#publish-self-contained 
That will require some work, however, to switch the project/solution to NetCore SDK, the code itself should be compatible (or require minor changes). It will also mean that we will need to produce a binary releases for every platform we want to support - linux-x64, macos-x64 etc

**Comment by nielshulstaert** [comment]
I've looked into the mono [mkbundle](https://www.mono-project.com/docs/tools+libraries/tools/mkbundle/) tool a while ago for an [issue](https://github.com/compomics/ThermoRawFileParser/issues/61). It seems that it's platform specific and it's not guaranteed that an ubuntu 18 executable will run on arch linux for example. 

I don't have experience with .NET core but I'll try to install it on my system. @caetera do you know the thermo libraries are compatible with .NET core?

**Comment by caetera** [comment]
> I don't have experience with .NET core but I'll try to install it on my system. @caetera do you know if the thermo libraries are compatible with .NET core?

Yes, for 5.0.38 there is a version complied for Net Standard 2.0 - i.e. NetCore 2.0 and higher -- https://docs.microsoft.com/en-us/dotnet/standard/net-standard

I have tried self-contained executables in other project, compiled on Ubuntu it worked on Arch without any problems. I did not perform extensive testing though.

**Comment by timosachsenberg** [comment]
It looks like the runtime is included in the package? Is this issue completed, or are still system/additional mono installations needed?

**Comment by caetera** [comment]
Hi @timosachsenberg,
no, mono is still a requirement. The `runtime` folder only contains the runtime versions of the `Mono.Unix` package (it is released that way).

**Comment by caetera** [comment]
This should be solved by the future release (#193) as well.

**Comment by timosachsenberg** [comment]
🙏

**Comment by ypriverol** [comment]
🤩 Thanks @caetera I already started working in the conda container.
